src/signatures/secp256k1_ecdsa.rs

```diff
@@ -38,20 +44,47 @@ impl SecretKey {
         k256::ecdsa::SigningKey::from_bytes(bytes.into())
             .map_err(|_| crate::Error::ConvertError {
                 from: "bytes",
-                to: "secp256k1 ecdsa secret key",
+                to: "Secp256k1 ECDSA secret key",
             })
             .map(Self)
     }
 
-    pub fn try_sign(&self, msg: &[u8]) -> crate::Result<Signature> {
+    /// Generate Secp256k1 ECDSA signature of message hash.
+    /// Signature generation can fail with a very low probability.
+    pub fn try_sign_prehash(&self, prehash: &[u8; PREHASH_LENGTH]) -> crate::Result<RecoverableSignature> {
+        self.0
+            .sign_prehash_recoverable(prehash)
+            .map_err(|_| crate::Error::SignatureError { alg: "Secp256k1 ECDSA" })
+            .map(|(sig, rid)| RecoverableSignature(Signature(sig), rid))
+    }
+
+    /// Generate Secp256k1 ECDSA signature of Keccak256 hash value of a message as used in Ethereum.
+    /// Signature generation can fail with a very low probability.
+    #[cfg(feature = "keccak")]
+    pub fn try_sign_keccak256(&self, msg: &[u8]) -> crate::Result<RecoverableSignature> {
+        let mut prehash = [0_u8; PREHASH_LENGTH];
+        keccak256(msg, &mut prehash);
+        self.try_sign_prehash(&prehash)
+    }
+
+    /// Secp256k1 ECDSA signature of Keccak256 hash value of a message as used in Ethereum.
+    #[cfg(feature = "keccak")]
+    pub fn sign_keccak256(&self, msg: &[u8]) -> RecoverableSignature {
```

Comment:
Why even have this function?

Reply:
I kinda agree, I'm not a big fan of functions that hide a panic for not much reason, maybe we should juste leave the user unwrap themselves. Or at least add documentation about the underlying panic.

Reply:
The probability of the event that signature generation fails is _very_ low. But it's not zero. Not clear how users should handle the error.

Reply:
Removed the functions.